### ISSUE TITLE
ignore does not work in glide.yaml

### ISSUE THREAD

**Issue by redhatlinux10** [Masterminds/glide#439]
in my codebase some `.go` file import `dis/auth/authn` package and this package exists in my project's codebase, and of cause i have added my project's codebase into `$GOPATH`

> e:\Workbench\DIS\auth\src>echo %GOPATH%
d:\GOPATH\EXT;e:\Workbench\DIS\auth\src\

here is my glide.yaml's part

> package: .
ignore:
\- dis/auth/authn
\- dis/auth/authz
\- dis/auth/mgo_session
\- dis/auth/server
import: 
\# ommit all imported packages

but when i do `glide update` in my project's codebase, there all some errors:

> e:\Workbench\DIS\auth\src>glide update
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for gopkg.in/yaml.v2.
[INFO] Fetching updates for github.com/docker/distribution.
[INFO] Fetching updates for github.com/golang/glog.
[INFO] Fetching updates for github.com/dchest/uniuri.
[INFO] Fetching updates for github.com/docker/libtrust.
[INFO] Fetching updates for github.com/deckarep/golang-set.
[INFO] Fetching updates for github.com/facebookgo/httpdown.
[INFO] Fetching updates for golang.org/x/crypto.
[INFO] Fetching updates for github.com/go-ldap/ldap.
[INFO] Fetching updates for github.com/syndtr/goleveldb.
[INFO] Fetching updates for gopkg.in/fsnotify.v1.
[INFO] Fetching updates for gopkg.in/mgo.v2.
[INFO] Resolving imports
[INFO] Fetching dis\auth\server into e:\Workbench\DIS\auth\src\vendor
**[ERROR] Error looking for dis\auth\server: Cannot detect VCS**
[INFO] Fetching updates for github.com/facebookgo/clock.
[INFO] Fetching updates for github.com/facebookgo/stats.
[INFO] Fetching golang.org/x/sys/unix into e:\Workbench\DIS\auth\src\vendor
[INFO] Fetching dis\auth\mgo_session into e:\Workbench\DIS\auth\src\vendor
**[ERROR] Error looking for dis\auth\mgo_session: Cannot detect VCS**
[INFO] Fetching gopkg.in/asn1-ber.v1 into e:\Workbench\DIS\auth\src\vendor
[INFO] Fetching dis\auth\authn into e:\Workbench\DIS\auth\src\vendor
**[ERROR] Error looking for dis\auth\authn: Cannot detect VCS**
[INFO] Fetching dis\auth\authz into e:\Workbench\DIS\auth\src\vendor
**[ERROR] Error looking for dis\auth\authz: Cannot detect VCS**
[INFO] Found Godeps.json file in vendor\github.com\docker\distribution
[INFO] Fetching updates for github.com/Sirupsen/logrus.
[INFO] Setting version for github.com/Sirupsen/logrus to 55eb11d21d2a31a3cc93838241d04800f52e823d.
[INFO] Setting version for github.com/docker/libtrust to fa567046d9b14f6aa788882a950d69651d230b21.
[INFO] Fetching updates for github.com/golang/snappy.
[INFO] Fetching updates for github.com/gorilla/mux.
[INFO] Setting version for github.com/gorilla/mux to e444e69cbd2e2e3e0749a2f3c717cec491552bbf.
[INFO] Fetching golang.org/x/net/context into e:\Workbench\DIS\auth\src\vendor
[INFO] Setting version for golang.org/x/net to 4876518f9e71663000c348837735820161a42df7.
[INFO] Fetching updates for github.com/gorilla/context.
[INFO] Setting version for github.com/gorilla/context to 14f550f51af52180c2eefed15e5fd18d63c0a64a.
[INFO] Downloading dependencies. Please wait...
[INFO] Setting references for remaining imports
[INFO] Project relies on 25 dependencies.
An Error has occurred

i have added these packages into ignore section, glide should ignore these packages(my own packages in my own codebase), right ?

please help.
glide:0.10.2

**Comment by mattfarina** [comment]
Two things:

First, If I understand it right you may not have a the `GOPATH` setup right. Given what I see I would expect:
```
- e:\Workbench\DIS\auth\src\
    |- dis\auth\authz
    |- dis\auth\authn
    |- dis\auth\mgo_session
    |- dis\auth\server
```

Your entire codebase needs to be inside the `GOPATH`. Have you read the docs on [organizing your code](https://golang.org/doc/code.html#Organization)? It's different from other languages.

Second, I think there is a bug in path separator handling. Ignoring should work even if your code organization has a problem.

**Comment by redhatlinux10** [comment]
yes,my directory structure is the same as what you expects.and i organize my code just as what https://golang.org/doc/code.html#Organization suggested. i just did not mention that.

but i think i did set `GOPATH` wrong. i should `set GOPATH=d:\GOPATH\EXT;e:\Workbench\DIS\auth\`
, not `set GOPATH=d:\GOPATH\EXT;e:\Workbench\DIS\auth\src`

and a question comes out. 
> `glide init` Scan a codebase and create a glide.yaml file containing the dependencies. 

which path does the codebase metioned before refer to  in my case ? `e:\Workbench\DIS\auth\` or `e:\Workbench\DIS\auth\src` ?
